live-build/misc/ansible-roles/appliance-build.minimal-common/tasks/main.yml

```diff
@@ -323,3 +323,37 @@
   with_items:
     - { key: 'kernel.core_pattern', value: '/var/crash/core.%e.%p.%t' }
     - { key: 'vm.overcommit_memory', value: '1' }
+
+#
+# Add the apt repos that have the debug versions of packages, so that we
+# can download a debug kernel to use with crash.
+#
+- apt:
+    name: 'ubuntu-dbgsym-keyring'
+- apt_repository:
+    repo: "{{ item }}"
+  with_items:
+    - deb http://ddebs.ubuntu.com bionic main restricted universe multiverse
+    - deb http://ddebs.ubuntu.com bionic-updates main restricted universe multiverse
+    - deb http://ddebs.ubuntu.com bionic-proposed main restricted universe multiverse
+
+#
+# Assume that the newest kernel is the one we are going to boot, and
+# fetch the symbols for that one.
+#
+- shell: ls /boot/vmlinuz-* | sed 's|/boot/vmlinuz-||' | sort -V | tail -n1
```

Comment:
I wonder if we could remove the “tail” call, and then process the output using “with_items” below.. so we could handle the case where multiple kernels are found.

I could test this out if it’s something we want to support.

Reply:
My thinking was that we would need a zfs kernel module for each kernel that we actually want to use, and since we only install one zfs module as part of the appliance build, we are only going to have one usable kernel, so we only need the debug symbols for that kernel.

However, instead of just choosing the newest kernel, it's probably better to either
 - Fail if the assumption that there is only one kernel installed doesn't hold, or
 - Just install symbols for every version like you mention, and let the other parts of the build worry about which and how many kernels to install.

I don't I really have a preference for one over the other.

Reply:
If the implementation for the latter doesn't make things more complicated, then I'd opt for that.

Although, doing what you have here should be fine.. if needed, we should have a check someplace else to fail when more than a single kernel is found.

Did you consider using `dpkg-query -Wf '${Version}' linux-image-generic` to generate the version, rather than `ls` and `sed`? I'm fine with your version, but am curious if `dpkg-query` didn't work or just wasn't tried.

Reply:
> Did you consider using dpkg-query -Wf '${Version}' linux-image-generic
Nvm.. Now after looking at the value returned by that command, it doesn't map directly to the package names, so we'd still have to massage it's output into the package that we want to install.

I'm good with what you have.

Reply:
The only reason I didn't use that was because it required a bit more parsing/formatting to get the output we need. It outputs something like `4.15.0.23.25`, and the string we need is of the form `4.15.0-23`. That shouldn't be too hard to do though, so I can change it if using `dpkg-query` is more correct.